src/core/exec-invoke.c

```diff
@@ -4816,6 +4850,23 @@ int exec_invoke(
                 }
         }
 
+        if (context->private_pids) {
+                if (ns_type_supported(NAMESPACE_PID)) {
+                        r = setup_private_pids(context, params);
+                        if (ERRNO_IS_NEG_PRIVILEGE(r))
+                                log_exec_warning_errno(context, params, r,
+                                                       "PrivatePIDs=yes is configured, but pid namespace setup failed, ignoring: %m");
+                        else if (r < 0) {
+                                *exit_status = EXIT_NAMESPACE;
+                                return log_exec_error_errno(context, params, r, "Failed to set up pid namespace: %m");
+                        }
+
+                        if (r > 0)
+                                return 0;
+                } else
+                        log_exec_warning(context, params, "PrivatePIDs=yes is configured, but the kernel does not support pid namespaces, ignoring.");
+        }
```

Comment:
and please add a big comment here, that clarifies that we are in a different process here suddenly. i.e. that before `private_pids` handling we had a different PID than we have now.